### PR TITLE
Set canonical url to custom variant

### PR DESCRIPTION
**Pull request by dsteenman**
# Checklist:

* [x] I have followed the guidelines in our [Contributing Guidelines](https://github.com/narative/gatsby-theme-novela/blob/master/CONTRIBUTING.md). _Especially our commitlint_
* [x] I have checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/fix/change.
* [x] I have checked for an open issue related to this. _There isn't one / There is one : #352 
* [x] I have performed a self-review of my own code.
* [x] I have performed the necessary tests locally.
* [ ] I have linted my code locally prior to submission.
* [x] If applicable, I have commented my code, particularly in hard-to-understand areas

# Type of PR

* [ ] Bug fix (_non-breaking change which fixes an issue_)
* [ ] New feature (_adding a feature following a feature-request issue_)
* [ ] Improvement (_improving an existing feature - includes `style:` and `perf:`commits_)
* [ ] Refactor (_rewriting existing code without any feature change_)
* [ ] (!) This change is or requires a documentation update

# Description

@aleccool213 I've updated this pr to make sure you can use custom canonical url's again. Sincs I wasn't able to fetch the pathnames from all pages (except articles), I think it's wiser for people to use this plugin: https://www.gatsbyjs.org/packages/gatsby-plugin-react-helmet-canonical-urls/ to enable canonical url's for their own pages

### CODE REVIEW

File: @narative/gatsby-theme-novela/src/components/SEO/SEO.tsx [hunk 1]
```diff
@@ -22,18 +22,18 @@ import Helmet from 'react-helmet';
 import { graphql, useStaticQuery } from 'gatsby';
 
 interface HelmetProps {
-  title: string;
+  authorName?: string;
```
Comment (dsteenman) [inline @narative/gatsby-theme-novela/src/components/SEO/SEO.tsx:25]: Sorted lines

File: @narative/gatsby-theme-novela/src/sections/article/Article.SEO.tsx [hunk 2]
```diff
@@ -49,17 +49,18 @@ const ArticleSEO: React.FC<ArticleSEOProps> = ({
...
-      description={article.excerpt}
-      image={imagelocation}
-      timeToRead={article.timeToRead}
-      published={article.date}
-      canonicalUrl={siteUrl + location.pathname.replace(/\/$/, "")}
-      dateforSEO={article.dateForSEO}
       authorName={authorsName}
-      authorsSlug={authorsSlug}
       authorsBio={authorsBio}
+      authorsSlug={authorsSlug}
```
Comment (dsteenman) [inline @narative/gatsby-theme-novela/src/sections/article/Article.SEO.tsx:54]: sorted lined

**Comment by dsteenman** [comment]
@aleccool213 please review

**Comment by brotzky** [comment]
Will give @aleccool213 some time. Once you've given it a look we can merge.

File: @narative/gatsby-theme-novela/src/components/SEO/SEO.tsx [hunk 3]
```diff
@@ -386,7 +391,7 @@ const SEO: React.FC<HelmetProps> = ({
       meta={metaTags}
     >
       <script type="application/ld+json">{schema}</script>
-      <link rel="canonical" href={canonicalUrl || site.siteUrl} />
+      {canonicalUrl && <link rel="canonical" href={canonicalUrl} />}
```
Comment (aleccool213) [inline @narative/gatsby-theme-novela/src/components/SEO/SEO.tsx:394]: Dope 👍

**Pull request merged**